beater/middleware/log_middleware.go

```diff
@@ -36,67 +36,97 @@ func LogMiddleware() Middleware {
 	return func(h request.Handler) (request.Handler, error) {
 
 		return func(c *request.Context) {
-			var reqID, transactionID, traceID string
-			tx := apm.TransactionFromContext(c.Request.Context())
-			if tx != nil {
-				// This request is being traced, grab its IDs to add to logs.
-				traceContext := tx.TraceContext()
-				transactionID = traceContext.Span.String()
-				traceID = traceContext.Trace.String()
-				reqID = transactionID
-			} else {
-				uuid, err := uuid.NewV4()
-				if err != nil {
-					id := request.IDResponseErrorsInternal
-					logger.Errorw(request.MapResultIDToStatus[id].Keyword, "error", err)
-					c.Result.SetWithError(id, err)
-					c.Write()
-					return
-				}
-				reqID = uuid.String()
+			args, err := requestArgs(c, logger.ECSEnabled())
+			if err != nil {
+				id := request.IDResponseErrorsInternal
+				logger.Errorw(request.MapResultIDToStatus[id].Keyword, "error", err)
+				c.Result.SetWithError(id, err)
+				c.Write()
+				return
 			}
-
-			reqLogger := logger.With(
-				"request_id", reqID,
-				"method", c.Request.Method,
-				"URL", c.Request.URL,
-				"content_length", c.Request.ContentLength,
-				"remote_address", utility.RemoteAddr(c.Request),
-				"user-agent", c.Request.Header.Get(headers.UserAgent))
-
-			if traceID != "" {
-				reqLogger = reqLogger.With(
-					"trace.id", traceID,
-					"transaction.id", transactionID,
-				)
-			}
-
-			c.Logger = reqLogger
+			c.Logger = logger.With(args...)
 			h(c)
 
 			if c.MultipleWriteAttempts() {
-				reqLogger.Warn("multiple write attempts")
+				c.Logger.Warn("multiple write attempts")
 			}
-
 			keyword := c.Result.Keyword
 			if keyword == "" {
 				keyword = "handled request"
 			}
-
-			keysAndValues := []interface{}{"response_code", c.Result.StatusCode}
-			if c.Result.Err != nil {
-				keysAndValues = append(keysAndValues, "error", c.Result.Err.Error())
-			}
-			if c.Result.Stacktrace != "" {
-				keysAndValues = append(keysAndValues, "stacktrace", c.Result.Stacktrace)
-			}
-
+			args = resultArgs(c, logger.ECSEnabled())
 			if c.Result.Failure() {
-				reqLogger.Errorw(keyword, keysAndValues...)
-			} else {
-				reqLogger.Infow(keyword, keysAndValues...)
+				c.Logger.Errorw(keyword, args...)
+				return
 			}
-
+			c.Logger.Infow(keyword, args...)
 		}, nil
 	}
 }
+
+func requestArgs(c *request.Context, ecsEnabled bool) ([]interface{}, error) {
```

Comment:
Instead of returning `[]interface` and using `Infow` and `Errorw` consider to return a logger and add the labels via `(*Logger).With`.